src/lib/stores/projects.store.ts

```diff
@@ -0,0 +1,10 @@
+import {readonly, writable} from 'svelte/store'
+import type {Project} from '$lib/types/project'
+
+const ProjectsStore = writable<Pick<Project, 'id' | 'name'>[]>([])
+
+export const setProjects = (projects: Project[]) => {
+    ProjectsStore.set(projects.map((p) => ({id: p.id, name: p.name})))
```

Comment:
А без map не канает?

```suggestion
    ProjectsStore.set(projects)
```